src/ADRIA.jl

```diff
@@ -56,19 +56,18 @@ include("ecosystem/corals/Corals.jl")
 include("ecosystem/const_params.jl")
 include("ecosystem/connectivity.jl")
 
+include("decision/CriteriaWeights.jl")
+
 include("Domain.jl")
 include("io/inputs.jl")
 
-include("decision/mcda_methods.jl")
-include("decision/Criteria.jl")
 include("decision/dMCDA.jl")
```

Comment:
Either move the include for `decision/dMCDA.jl` into the submodule or group the includes together - which ever is most appropriate.

Reply:
I think I put them in this order because `CriteriaWeights.jl` is needed in `Domain.jl` and `Domain.jl` is needed in `dMCDA.jl`.

Currently the submodule `decision` is inside `dMCDA.jl`. Should I instead make a separate file called `decision.jl` which contains the module and all the imorts?

Reply:
> I think I put them in this order because `CriteriaWeights.jl` is needed in `Domain.jl` and `Domain.jl` is needed in `dMCDA.jl`.

This is a smell that suggests you need to rethink the structure then.

Why is `domain` needed in `dMDCA`? Can you separate out the dependency?

Reply:
The Domain type is used to extract parameters used in the `DMCDA_vars` constructor. This includes `sim_constants`, `TP_data`, `median_site_distance` and `site_distances`. I could make these separate inputs but it would mean increasing the number of input arguments. Should I do this?

Reply:
Hang on - `DMCDA_vars` essentially acts as a data container. 

Do we need that at all?

If we do need it, everything is being shoved into it and only some of the data is relevant to each of the steps/methods that take it as an argument. Could you not split that up to something more appropriate and smaller?

Reply:
Okay, it seems you've uncovered a big smell that points to a required redesign. 
It's going to be too big for this PR so leave things be.

**Add a comment that the include order has to be like it is and why.**

Reply:
The plan in the changes for making the MCDA more generic was come up with an alternative to MCDA_vars because it is a veryt inefficient way of doing things. I think in future changes the scenario struct itself should be passed in (as MCDA_vars is predominantly a container criteria weights), with a NamedDimsArray for criteria (essentially the decision matrix as a NamedDimsArray) for storing criteria values. But agree this is part of a larger problem.